src/fault.h

```diff
@@ -0,0 +1,71 @@
+#ifndef FAULT_H
+#define FAULT_H
+
+#include "string_id.h"
+#include "json.h"
+
+class fault;
+using fault_id = string_id<fault>;
+
+class Skill;
+using skill_id = string_id<Skill>;
+
+using quality_id = std::string;
+
+using itype_id = std::string;
+
+class fault
```

Comment:
Yes, including (for example) alternative requirements. If/when that class gets extended to provide more configuration (like *consumption* of items with specific quality), your repair action can use it without any modification.

And if you look close: the survivor *is* crafting: take an item with faults + some tools + some items yields an item with less faults. Looks like crafting to me.